File: eternal_return/src/main/java/org/EternalReturn/EREntity/GlobalMonobehav/InfinityHealing.java
package org.EternalReturn.EREntity.GlobalMonobehav;

import org.EternalReturn.EREntity.EREntityMonobehaviour;
import org.EternalReturn.EREntity.Event.EREntityDamagedEvent;
import org.EternalReturn.Util.dpengine.behaviour.MonobehaviourEvent;
import org.bukkit.entity.Entity;
import org.bukkit.entity.LivingEntity;

import java.util.Collection;

public class InfinityHealing extends EREntityMonobehaviour<EREntityDamagedEvent> {

    @Override
    public void start(EREntityDamagedEvent event) {
        Entity entity = this.getEntity();
        if(entity instanceof LivingEntity livingEntity){
            livingEntity.setHealth(20);
        }
    }

    @Override
    public void update(Collection<MonobehaviourEvent> eventList) {

    }
}

File: eternal_return/src/main/kotlin/org/eternalreturn/erentity/globalmonobehav/InfinityHealing.kt
package org.eternalreturn.erentity.globalmonobehav

import org.eternalreturn.erentity.EREntityMonobehaviour
import org.eternalreturn.erentity.events.EREntityDamagedEvent
import org.eternalreturn.util.dpengine.behaviour.MonobehaviourEvent
import org.bukkit.entity.LivingEntity

class InfinityHealing : EREntityMonobehaviour<EREntityDamagedEvent>() {
    public override fun start(event: EREntityDamagedEvent) {
        val entity = this.entity
        if (entity is LivingEntity) {
            entity.setHealth(20.0)
        }
    }

    public override fun update(eventMap: Map<Class<out MonobehaviourEvent>,MonobehaviourEvent>) {
    }
}
